### ISSUE TITLE
terse way to mention PR or issues by `#<number>`

### ISSUE THREAD

**Issue by scarf005** [statch/gitbot#269]
**Is your feature request related to a problem? Please describe.**
as `git pr <number>` is verbose, i find it difficult to mention multiple PRs in compact manner.

`git pr 2515`
![image](https://user-images.githubusercontent.com/54838975/235456548-01827f0b-2d24-4ca4-8af4-8fd5270cef4e.png)

**Describe the solution you'd like**
a new hash command like that of github to display issue/PR in terse manner like: 

`git #2515`
![image](https://user-images.githubusercontent.com/54838975/235456712-d5f933fb-7857-4967-9557-18fa8b3b8c1a.png)

#### command
- `git <user>/<repo> #<number>` 
- `git #<number>` (repo given in config)

**Comment by seven7ty** [comment]
Hi there! 
I've had something similar in mind for a while now, so this may be coming to GitBot this week/next week.
That said, I don't quite understand what you mean by "mentioning multiple PRs in compact manner."
Do you mean you would like the new, short-form command (i.e. `git #number`) to accept multiple PRs/issues,
as in `git #123 #456 #789`?

**Comment by scarf005** [comment]
that would also work. what i had in mind for "mentioning multiple PRs in compact manner." was something like which fully fills an window:
```
git #123
git #456
git #789
```

**Comment by seven7ty** [comment]
I see. Definitely will consider this in the next update, although I'm not sure what it will be -- v4, or something earlier. 
Let's keep the issue open for the sake of this remaining active. Cheers!

**Comment by scarf005** [comment]
i think this feature could be implemented by copy-pasting `git pr command`, replace `pr` to `#`, and only keeping title and removing all others. in that case, i could open a PR.

**Comment by seven7ty** [comment]
Feel free to experiment, although what you suggest would only work if called `git # 1456` and not `git #456`. 

I haven't done anything in regard to this, but I'm pretty sure it would need a separate handler that is independent of the internal discord.py `commands.Bot` command parser (since the `#number` is not a name of a command but rather an argument in place of it). 

Off the top of my head I'd probably just handle it inside `errors.py` in the "command does not exist" section - since that scope ensures that it was not any valid command, but the prefix was there (hence invocation, but failed). This would add very little overhead and wouldn't be too much of a hassle to implement if thought through well.

That said, I do encourage you to experiment on your own since I might be missing something here.

**Comment by seven7ty** [comment]
I just realized I might've misunderstood what you want to achieve with this command. Do you just want a one-line breakdown of the title of the PR/issue?

In that case, it could be running either on a persistent basis (the bot looking for messages containing `#number(s)` inside them and responding in the format you suggest) or perhaps on a context menu basis (message mentioning `#number(s)` can be right clicked and then they get extracted and inspected as you say)

This approach is a little more complex but interesting nonetheless.

**Comment by scarf005** [comment]
> Do you just want a one-line breakdown of the title of the PR/issue?

yes. example usage from another discord bot:
![image](https://user-images.githubusercontent.com/54838975/235680019-4ff75aa2-6090-4e46-846f-16296fe1d325.png)

former one looks easier to implement, so i'll check them. some pointers to where to look would be appreciated.

**Comment by seven7ty** [comment]
In terms of the "message-scanning" approach, the best place to do something like this would be calling a handler inside `cogs/backend/handle/events/events.py` that would first be implemented in `cogs/backend/handle/events/_event_tools.py`.

From there, inside the handler itself, you could get the message author's quick-access repo, extract issue and PR numbers, then query them one by one (probably using a new GraphQL query utilizing `issueOrPullRequest` from GitHub's schema) and just respond if everything worked okay. A bit of a chore, but nothing that hasn't been implemented before.

**Comment by scarf005** [comment]
...oh. I've looked into it, but the code seems fairly complex for me. i'm not confident i could implement this on my own without getting significant help. could you please implement the PR instead?

**Comment by seven7ty** [comment]
No worries! The code is pretty hard to follow if you're just getting started with it, so I totally get what you mean. 
I'll implement the PR along with some other updates in the coming weeks, cheers!
